Use `String.prototype.replaceAll()` where appropriate

Snuffleupagus opened the pull request:
This fairly new method allows replacing *multiple* occurrences within a string without having to use regular expressions.

Please refer to:
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#browser_compatibility

Snuffleupagus committed:
Use `String.prototype.replaceAll()` where appropriate

This fairly new method allows replacing *multiple* occurrences within a string without having to use regular expressions.

Please refer to:
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#browser_compatibility

Snuffleupagus commented:
/botio test

calixteman commented on external/importL10n/locales.js:
```diff
@@ -27,7 +27,7 @@ const DEFAULT_LOCALE = "en-US";
 
 function normalizeText(s) {
-  return s.replace(/\r\n?/g, "\n").replace(/\uFEFF/g, "");
+  return s.replace(/\r\n?/g, "\n").replaceAll("\uFEFF", "");
```
Why don't you replace the first `replace` by a `replaceAll` ?
I guess it doesn't matter that much but at least it's more explicit than a `g` regex.

Snuffleupagus replied:
Do you mean simply changing `replace` into `replaceAll`, or something else?

Note that I don't *think* that we can easily replace the regular expression with a string, considering what it does. Unless I'm missing something the replacement currently does the following, which can't be captured with just a string:
 - `\r` -> `\n`
 - `\r\n` -> `\n`

calixteman replied:
`replaceAll` can take a regex but I missed in the docs that it has to have a `g`...
But as said, it'd slightly improve the readability of the code, that said it's an opinion, so do whatever you want with it.

calixteman replied:
Yep it's fine, r=me.
And `replaceAll` throws when the `g` is missing so it could help to avoid to overlook it.

calixteman reviewed (approved):
LGTM.

Snuffleupagus committed:
Enable the `unicorn/prefer-string-replace-all` ESLint plugin rule

Note that the `replaceAll` method still requires that a *global* regular expression is used, however by using this method it's immediately obvious when looking at the code that all occurrences will be replaced; please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#parameters

Please find additional details at https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-replace-all.md

Snuffleupagus commented:
*Hopefully the bots use a modern enough Node.js version for this to work...*

/botio preview

Snuffleupagus commented:
/botio test
